test-data/unit/check-enum.test

```diff
@@ -2681,3 +2681,50 @@ reveal_type(Wrapper.Nested.FOO)  # N: Revealed type is "Literal[__main__.Wrapper
 reveal_type(Wrapper.Nested.FOO.value)  # N: Revealed type is "builtins.ellipsis"
 reveal_type(Wrapper.Nested.FOO._value_)  # N: Revealed type is "builtins.ellipsis"
 [builtins fixtures/enum.pyi]
+
+[case testEnumItemsEqualityToLiterals]
+# flags: --python-version=3.11 --strict-equality
+from enum import Enum, StrEnum, IntEnum
+
+class A(str, Enum):
+    a = "b"
+    b = "a"
+
+A.a == "a"
```

Comment:
It's not clear to me why `A.a == "a"` should be allowed

Reply:
Yep, that's basically what I want to gather feedback on: at runtime it's true, so there is some merit in allowing such comparisons (see also #17162 for exactly such example).

I'll need to update all added tests to check reachability - if the comparison is not considered non-overlapping, corresponding `if` bodies must be reachable.

Reply:
Did I misunderstand something?
```pycon
>>> import enum
>>> class X(enum.Enum):
...   a = "b"
... 
>>> X.a == "a"
False
```

Reply:
Ough, my bad, that part of #17162 is based on a false premise, plain `enum.Enum` doesn't compare equal, of course.

In the test case, however, I'm checking `StrEnum` and `str, Enum` subclasses, and they *do* compare equal to strings matching the values:

```python
import enum

class A(str, enum.Enum):
    A = 'a'

A.A == 'a'  # True
```

Reply:
(so in #17162 both cases must be reachable, and IMO shouldn't show `comparison-overlap` errors)

Reply:
I think we might be talking about different things. I'm referring to an `X.a` which is a `"b"`, not a `"a"`. I somehow missed that it doesn't return `True` for non-string-enums though. For reference:

```pycon
>>> import enum
>>> class X(str, enum.Enum):
...   a = "b"
... 
>>> X.a == "a"  # this is currently allowed, but returns False
False
>>> X.a == "b"  # this is currently disallowed, but returns True
True
```

While this PR handles the second case, I think the first case matters too. (Though I haven't cared enough to look at mypy internals and see why it happens ATM)

Reply:
IMO, being more smart is good, especially since I doubt it's that many extra lines?

I agree many people may not directly compare literals by value, but I imagine it's useful for like type narrowing. (since we treat enums as a union of their literal elements, I think?)